build/lib/src/asset/id.dart

```diff
@@ -110,9 +110,9 @@ class AssetId implements Comparable<AssetId> {
   /// Note that this is intended for communicating ids across isolates and not
   /// for persistent storage of asset identifiers. There is no guarantee of
   /// backwards compatibility in serialization form across versions.
-  AssetId.deserialize(data)
-      : package = data[0],
-        path = data[1];
+  AssetId.deserialize(List data)
```

Comment:
Will update the comment. I think `List<String>` would work today be cause `<dynamic>[] is List<String>` passes - but I'm not sure if that will always be the case when the VM starts using strong mode semantics at runtime. I think that what I have is more indicative of how the checks work - each element that gets pulled out of the List needs to be cast to a String because each element could fail at runtime.

On a related note - I wonder if we should pull serialization out of these classes...